shared/src/main/java/uk/gov/di/authentication/shared/entity/UserProfile.java

```diff
@@ -160,4 +181,39 @@ public UserProfile setSalt(byte[] salt) {
         this.salt = ByteBuffer.wrap(salt);
         return this;
     }
+
+    @Override
+    public Map<String, AttributeValue> toItem() {
+        Map<String, AttributeValue> attributes = new HashMap<>();
+        if (getEmail() != null) attributes.put(ATTRIBUTE_EMAIL, new AttributeValue(getEmail()));
+        if (getSubjectID() != null)
+            attributes.put(ATTRIBUTE_SUBJECT_ID, new AttributeValue(getSubjectID()));
+        attributes.put(
+                ATTRIBUTE_EMAIL_VERIFIED,
+                new AttributeValue().withN(isEmailVerified() ? "1" : "0"));
+        if (getPhoneNumber() != null)
+            attributes.put(ATTRIBUTE_PHONE_NUMBER, new AttributeValue(getPhoneNumber()));
+        attributes.put(
+                ATTRIBUTE_PHONE_NUMBER_VERIFIED,
+                new AttributeValue().withN(isPhoneNumberVerified() ? "1" : "0"));
+        if (getCreated() != null)
+            attributes.put(ATTRIBUTE_CREATED, new AttributeValue(getCreated()));
+        if (getUpdated() != null)
+            attributes.put(ATTRIBUTE_UPDATED, new AttributeValue(getUpdated()));
+        if (getTermsAndConditions() != null)
+            attributes.put(
+                    ATTRIBUTE_TERMS_AND_CONDITIONS, getTermsAndConditions().toAttributeValue());
+        if (getClientConsent() != null) {
+            Collection<AttributeValue> consents = new ArrayList<>();
+            getClientConsent().stream().forEach(c -> consents.add(c.toAttributeValue()));
```

Comment:
You don't need the `stream` here so it can be ` getClientConsent().forEach(c -> consents.add(c.toAttributeValue()));`

Reply:
updated